u-mode/src/cert.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2023 by Rivos Inc.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+extern crate libuser;
+use libuser::*;
+
+use der::Decode;
+use ed25519::Signature;
+use ed25519_dalek::Signer;
+use generic_array::GenericArray;
+use rice::x509::certificate::{Certificate, MAX_CERT_SIZE};
+use rice::x509::extensions::dice::tcbinfo::DiceTcbInfo;
+use rice::x509::request::CertReq;
+use rice::x509::MAX_CSR_LEN;
+use u_mode_api::cert::*;
+
+#[derive(Debug)]
+pub enum Error {
+    /// Input CSR buffer size too small.
+    CsrBufferTooSmall(usize, usize),
+    /// Cannot parse CSR.
+    CsrParseFailed(der::Error),
+    /// Cannot verify CSR.
+    CsrVerificationFailed(rice::Error),
+    /// Cannot add FWID extension.
+    FwidAddFailed(rice::Error),
+    /// Could not create TcbInfo Extensions.
+    TcbInfoFailed(rice::Error),
+    /// Cannot create Certificate.
+    CertificateCreationFailed(rice::Error),
+    /// Output Certificate buffer too small.
+    CertificateBufferTooSmall(usize, usize),
+}
+
+struct UmodeSigner {}
+
+impl Signer<Signature> for UmodeSigner {
+    fn try_sign(&self, _: &[u8]) -> Result<Signature, ed25519::Error> {
+        // TODO: Implement Signing of certificate.
+        Signature::from_bytes(&[0; 64])
+    }
+}
+
+pub fn get_certificate_sha384(
+    csr_input: &[u8],
+    data: GetEvidenceShared,
+    certout: &mut [u8],
```

Comment:
Ack for `evidence`. 

Disagree on `cert_bytes`, which is used later in the function. the subtlety here is that `csr_input` and `certout` are guest mapped. So they need a copy. I don't want to have this copy in `src/main.rs` because I want to keep all rice-specific knowledge in this file.

So possibly `csr_guest` and `cert[out]_guest` could be better names.

Reply:
Renamed to `evidence`, changed `certout` to `cert_output` to be more simmetrical with input, and fixed the comments.